lib/text_count_util.ex

```diff
@@ -3,16 +3,33 @@ defmodule TextCountUtil do
   Documentation for `TextCountUtil`.
   """
 
+  alias TextCountUtil.Counter
+
   @doc """
-  Hello world.
+  Calculate words count of text
 
   ## Examples
 
-      iex> TextCountUtil.hello()
-      :world
+      iex> TextCountUtil.words_count("./test/fixtures/example.txt")
+      5
 
   """
-  def hello do
-    :world
+  @spec words_count(String.t()) :: non_neg_integer()
+  def words_count(filename) do
+    File.stream!(filename)
+    |> Stream.with_index(1)
+    |> Stream.map(&(do_count(&1)))
+    |> Stream.map(&(get_count(&1)))
+    |> Enum.sum
+  end
+
+  defp do_count({line, idx}) do
+    {:ok, pid} = Counter.start_link(idx)
+    GenServer.cast(pid, {:do_count, line})
```

Comment:
This is the only action this process is able to perform, hence I’d do it directly explicitly in [`handle_continue/2`](https://hexdocs.pm/elixir/GenServer.html#c:handle_continue/2) (see below,) without a dedicated message.